dali/python/nvidia/dali/external_source.py

```diff
@@ -18,9 +18,10 @@ def _get_batch_shape(data):
 
 def _check_data_batch(data, batch_size, layout):
     shape, uniform = _get_batch_shape(data)
-    if len(shape) != batch_size:
+    if len(shape) > batch_size:
```

Comment:
I think this change should work until we enable it for the users (and add tests).